tests/e2e/utils/KubernetesCommandLineToolsExecutor.ts

```diff
@@ -0,0 +1,131 @@
+        Logger.debug(`${this.getLoggingName(this.getWorkspacePodName.name)}: Get workspace pod name.`);
+        const output: ShellString = this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} get pod -l controller.devfile.io/devworkspace_name=${this.workspaceName} -n ${this.namespace} -o name`);
+        return output.stderr ? output.stderr : output.stdout.replace('\n', '');
+    }
+
+    deleteDevWorkspace(): void {
+        Logger.debug(`${this.getLoggingName(this.deleteDevWorkspace.name)}: Delete '${this.workspaceName}' workspace`);
+        this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} patch dw ${this.workspaceName} -n ${this.namespace} -p '{ "metadata": { "finalizers": null }}' --type merge || true`);
+        this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} delete dw ${this.workspaceName} -n ${this.namespace} || true`);
+    }
+
+    applyAndWaitDevWorkspace(yamlConfiguration: string): ShellString {
+        if (this.KUBERNETES_COMMAND_LINE_TOOL === KubernetesCommandLineTool.KUBECTL) {
+            this.createNamespace();
+        }
+        this.apply(yamlConfiguration);
+        const output: ShellString = this.waitDevWorkspace();
+        KubernetesCommandLineToolsExecutor.pod = this.getWorkspacePodName();
+        KubernetesCommandLineToolsExecutor.container = this.getContainerName();
+        return output;
+    }
+
+    executeCommand(commandToExecute: string): ShellString {
+        Logger.debug(`${this.getLoggingName(this.executeCommand.name)}:`);
+        return this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} exec -i ${KubernetesCommandLineToolsExecutor.pod} -n ${this.namespace} -c ${KubernetesCommandLineToolsExecutor.container} -- sh -c "${commandToExecute}"`);
+    }
+
+    apply(yamlConfiguration: string): void {
+        Logger.debug(`${this.getLoggingName(this.apply.name)}:`);
+        this.execWithLog(`cat <<EOF | ${this.KUBERNETES_COMMAND_LINE_TOOL} apply -n ${this.namespace} -f - \n` +
+            yamlConfiguration + '\n' +
+            'EOF');
+    }
+
+    waitDevWorkspace(timeout: number = 360): ShellString {
+        Logger.debug(`${this.getLoggingName(this.waitDevWorkspace.name)}: Wait till workspace ready.`);
+        return this.execWithLog(`${(this.KUBERNETES_COMMAND_LINE_TOOL)} wait -n ${this.namespace} --for=condition=Ready dw ${this.workspaceName} --timeout=${timeout}s`);
+    }
+
+    createNamespace(): void {
+        Logger.debug(`${this.getLoggingName(this.createNamespace.name)}: Create namespace "${this.namespace}".`);
+        this.execWithLog(`${this.KUBERNETES_COMMAND_LINE_TOOL} create namespace ${this.namespace}`);
+    }
+
+    private getLoggingName(methodName: string): string {
+        return `${this.constructor.name}.${methodName} - ${(this.KUBERNETES_COMMAND_LINE_TOOL)}`;
+    }
+
+    private setNamespace(_namespace: string | undefined): string {
+        return _namespace !== undefined ? _namespace : TestConstants.TS_SELENIUM_OCP_USERNAME + '-devspaces';
```

Comment:
user namespace has prefix `-che` in case of Eclipse Che

Reply:
it is not minikube specific

There will be `<username>-che` user namespace used to create workspaces in Eclipse Che deployed to both OpenShift and minikube.

Reply:
Anyway, `-devspaces` name is misleading in eclipse/che repo.

Ideally, it would be better to use the same namespace which Dashboard is using when list and create workspaces.

Reply:
What do you think about adding dedicated parameter ` TS_API_TEST_NAMESPACE` to define namespace name explicitly when run the test against Dev Spaces, with default value `TS_SELENIUM_OCP_USERNAME + '-che'` ?

Reply:
added TS_API_TEST_NAMESPACE and updated method. it should set namespace according to base url. also you can sat it manualy